cmd/shortener/repositories/postgres.go

```diff
@@ -3,74 +3,65 @@ package repositories
 import (
 	"context"
 	"errors"
+	"github.com/jackc/pgx/v4/pgxpool"
 	"log"
 	"time"
 
 	"github.com/2heoh/yap_url_shortener/cmd/shortener/entities"
 	"github.com/jackc/pgconn"
-	"github.com/jackc/pgx/v4"
 )
 
 var (
 	ErrDBConnection = errors.New("no DB connection")
 	ErrKeyExists    = errors.New("key exists")
 )
 
-//type KeyExistsError struct {
-//	Key string
-//	Err error
-//}
-//
-//func (k *KeyExistsError) Error() string {
-//	return fmt.Sprintf("%s: %v", k.Err, k.Key)
-//}
-//
-//func NewKeyExistsError(key string, err error) error {
-//	return &KeyExistsError{Key: key, Err: err}
-//}
-
-const timeout = 5 * time.Second
+const (
+	timeout = 5 * time.Second
+)
 
 type DBRepository struct {
-	connection *pgx.Conn
+	connection *pgxpool.Pool
 }
 
-func (r *DBRepository) AddBatch(urls []entities.URLItem, userID string) ([]entities.ShortenURL, error) {
-	var result []entities.ShortenURL
+func NewDatabaseRepository(dsn string) Repository {
 
-	ctx := context.Background()
-	tx, err := r.connection.Begin(ctx)
-	if err != nil {
-		return nil, err
+	repo := &DBRepository{
+		connection: connect(dsn),
 	}
 
-	defer tx.Rollback(ctx)
+	repo.init()
 
-	for _, item := range urls {
-		_, err = tx.Exec(ctx, "insert into links (userid, key, url) values ($1, $2, $3)", userID, item.CorrelationID, item.OriginalURL)
-		if err != nil {
-			return nil, err
-		}
-		result = append(result, entities.ShortenURL{Key: item.CorrelationID})
-	}
+	return repo
+}
 
-	err = tx.Commit(ctx)
-	if err != nil {
+func (r *DBRepository) MakeDelete(candidate entities.DeleteCandidate) error {
+	log.Printf("\\_/   %v \n", candidate)
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
 
-		return nil, err
+	sql := "UPDATE  links SET deleted=true WHERE key=$1 and userid=$2"
```

Comment:
а здесь для batch delete можно было бы сделать примерно так:
```suggestion
	sql := "UPDATE  links SET deleted=true WHERE key in ($1) and userid=$2"
	// тогда sql на выходе будет примерно следующий:
	// UPDATE  links SET deleted=true WHERE key in (id1, id2) and userid=2
```
В таком случае пометка всех ссылок как удаленные произойдет за один запрос к базе, что будет быстрее, так как не будет тратиться время на межсетевое взаимодействие как минимум)